src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryService.java

```diff
@@ -208,12 +206,12 @@ private PermissionRequirements findPermissionRequirement(SearchTaskRequest searc
                                                              boolean isGranularPermissionEnabled) {
         if (isGranularPermissionEnabled) {
             //When granular permission feature flag is enabled, request is expected only in new format
-            SearchParameterRequestContext context = extractRequestContext(searchTaskRequest);
+            RequestContext context = searchTaskRequest.getRequestContext();
             if (context == null) {
                 return PermissionRequirementBuilder.builder().buildSingleType(READ);
-            } else if (context.isEqual(RequestContext.AVAILABLE_TASKS)) {
+            } else if (context.equals(RequestContext.AVAILABLE_TASKS)) {
                 return PermissionRequirementBuilder.builder().buildSingleRequirementWithAnd(OWN, CLAIM);
-            } else if (context.isEqual(RequestContext.ALL_WORK)) {
+            } else if (context.equals(RequestContext.ALL_WORK)) {
                 return PermissionRequirementBuilder.builder().buildSingleType(MANAGE);
             }
             return PermissionRequirementBuilder.builder().buildSingleType(READ);
```

Comment:
This is probably something to change in a follow up ticket.  Should be throwing a BadRequest problem response if they provide a string that's not one of the ones we expect.